src/Components/WebAssembly/WebAssembly/src/Services/WebAssemblyNavigationManager.cs

```diff
@@ -33,6 +33,12 @@ protected override void EnsureInitialized()
             Initialize(baseUri, uri);
         }
 
+
+        public void InitializeWithParameters(string baseUri, string uri)
+        {
+            Initialize(baseUri, uri);
```

Comment:
If this is being done here, are there also updates to the `EnsureInitialized` method above? Seems wrong that it also calls `Interop.GetBaseUri`.

Overall, the initialization flow starts to seem a bit complicated as there's more than one code path that calls these initialization methods. If there any way to consolidate this that would be good.

For example, it's not clear to me why the `InitializeNavigationManager` method now exists in `WebAssemblyHostBuilder`. Since the only thing it does is use synchronous interop to fetch its config from the JS side, would it work more simply to have a regular constructor on `WebAssemblyNavigationManager` that takes a `WebAssemblyJSRuntimeInvoker` and provides its own initialization logic? If so, the `EnsureInitialized` can become a no-op, and just contain a comment saying we have nothing else to do, and we wouldn't need this `InitializeWithParameters` method. The whole flow would then be much more understandable.

I haven't really thought through all the details so maybe there are reasons it has to work like it does, but just want to check.

Reply:
Yeah, this was kind of a bit of a muck before wasn't it? I've spruced it up a bit and moved the initialization logic back inside the NavigationManager.

@pranavkm had talked about moving the logic to the startup code. Perhaps my original implementation didn't reflect his intent here but, IMO. it reads cleaner to have all the initialization logic within the NavigationManager itself.

Reply:
Yeah, I thought it'd be a lot easier to read if all of interop calls were clustered as part of the host initialization. 

```C#
// WebHostBuilder.cs

var baseUri = ...
var uri = ...

hostEnvironment.BaseUri = baseUri;

// need the static for JSInterop
WebAssemblyNavManager.Instance = new WebAssemblyNavManager(baseUri, uri);
```

Assigning a static property from within the ctor feels a bit strange.

Reply:
I spruced this up a bit. Let me know what you think.

Reply:
This looks great now!